run.c

```diff
@@ -573,10 +573,10 @@ Cell *intest(Node **a, int n)	/* a[0] is index (list), a[1] is symtab */
 		nsub = strlen(getsval(subseploc));
 		if (!adjbuf(&buf, &bufsz, strlen(buf)+strlen(s)+nsub+1, recsize, 0, "intest"))
 			FATAL("out of memory deleting %s[%s...]", x->nval, buf);
-		strcat(buf, s);
+		strlcat(buf, s, bufsz);
 		tempfree(x);
 		if (p->nnext)
-			strcat(buf, *SUBSEP);
+			strlcat(buf, *SUBSEP, bufsz);
```

Comment:
`strlcat` is a BSD-specific function.  Please rework by either adding specific checks inline and using plain `strcat`, or by including the code for strlcat itself in the file.  Thanks!

Reply:
fixed in a much better way.